src/routes/+layout.server.ts

```diff
@@ -11,39 +12,59 @@ export const load = async ({
   fetch,
   depends,
   locals: { contentfulClient, currentUser, previewAuthenticationError },
+  url: { pathname },
 }) => {
   // prod indicator is sent with page data and is currently used to determine:
   //   * whether we should connect to Vercel Speed Insights
   //   * if we should set up GA in debug mode or prod mode
   const isProd = VERCEL_ENV === "production";
+
   const clientCurrentUser: User | undefined = currentUser
     ? { name: currentUser.name, email: currentUser.email, avatarURL: currentUser.avatarURL }
     : undefined;
-  const pageMetadataMapPromise = loadPageMetadataMap({ contentfulClient });
-  const headerPrimaryNavItemsPromise = pageMetadataMapPromise.then(({ pageMetadataMap }) =>
-    loadMainNav({ pageMetadataMap, contentfulClient }),
-  );
-  const footerNavItemsPromise = pageMetadataMapPromise.then(({ pageMetadataMap }) =>
-    loadFooterNav({ pageMetadataMap, contentfulClient }),
-  );
-  const sideNavMapPromise = Promise.all([
-    pageMetadataMapPromise,
-    headerPrimaryNavItemsPromise,
-  ]).then(([{ pageMetadataMap }, navItems]) => loadSideNavMap(pageMetadataMap, navItems));
+
+  const { pageMetadataMap, pathsToIDs } = await loadPageMetadataMap({ contentfulClient });
+
+  // As soon as we have the requisite data, determine if we need to redirect
+  //   the user (whether internally or externally).
+  const metadataID = pathsToIDs.get(pathname);
+  if (metadataID) {
+    const pageMetadata = pageMetadataMap.get(metadataID);
+    if (pageMetadata) {
+      if (pageMetadata.internalRedirect) {
+        const internalRedirectPageMetadata = pageMetadataMap.get(
+          pageMetadata.internalRedirect.sys.id,
+        );
+        if (internalRedirectPageMetadata?.url) {
+          throw redirect(301, internalRedirectPageMetadata.url);
```

Comment:
`301` felt like the right option here for the ones available, but let me know if you think there's a better option.